File: src/ders8/rpgornekjava/Paladin.java
package ders8.rpgornekjava;

public class Paladin extends Oyuncu implements BuyuYapabilir, KalkanKullanabilir{

    public Paladin(String isim, int saglik) {
        super(isim, saglik);
    }

    public Paladin(String isim) {
        super(isim);
        this.saglik = 70;
    }

    @Override
    public void buyuYap() {
        System.out.println("Paladin büyü yapıyor!");
    }

    @Override
    public void kalkanKullan() {
        System.out.println("Paladin kalkan kullanıyor!");
    }
}

File: src/ders8/rpgornekkotlin/Paladin.kt
package ders8.rpgornekkotlin

class Paladin(isim: String, saglik: Int = 70) : Oyuncu(isim, saglik), BuyuKullanabilir, KalkanKullanabilir {
    override fun kalkanKullan() {
        println("$isim kalkan kullandı!")
    }

    override fun buyuYap() {
        println("$isim büyü yapıyor...")
    }
}